src/main/java/org/jabref/logic/util/strings/EmacsStringManipulator.java

```diff
@@ -0,0 +1,178 @@
+package org.jabref.logic.util.strings;
+import org.jabref.model.util.ResultingEmacsState;
+
+public class EmacsStringManipulator {
+    private enum LetterCase {
+        UPPER,
+        LOWER,
+        CAPITALIZED
+    }
+
+    private enum Direction {
+        NEXT, PREVIOUS
+    }
+
+    private static ResultingEmacsState setNextWordsCase(String text, int pos, LetterCase targetCase) {
+        StringBuilder res = new StringBuilder();
+
+        boolean firstLetter = true;
+        int i = pos;
+        boolean firstLoop = true;
+        StringBuilder newWordBuilder = new StringBuilder();
+        for (; i < text.length(); i++) {
+            // Swallow whitespace
+            while (firstLoop && i < text.length() && !String.valueOf(text.charAt(i)).matches("\\w")) {
```

Comment:
As you are using the `\\w` regex here multiple times it makes sense to compile it and use Pattern.compile(...) with a matcher then.